### PR TITLE
[RFC][UN-14455] Fix tooltip on scroll

### PR DESCRIPTION
**Pull request by niceoutput**
This PR is intended to solve the issue below
https://uniplaces.atlassian.net/browse/UN-14455

### CODE REVIEW

File: addon/components/uni-tooltip.js [hunk 1]
```diff
@@ -31,6 +31,21 @@ export default Component.extend(ClickOutsideMixin, {
     this._hideTooltip();
   },
 
+  didInsertElement() {
+    let clickHandler = this.get('handleOutsideClick').bind(this);
+
+    document.addEventListener('scroll', clickHandler, {
+      capture: true,
```
Comment (mmelo) [inline addon/components/uni-tooltip.js]: this fits into a single line mate 👍

**Review by mmelo — APPROVED** [review]

File: addon/mixins/click-outside.js [hunk 2]
```diff
@@ -17,12 +17,12 @@ export default Mixin.create({
   setupOutsideClickListener: on('didInsertElement', function() {
     let clickHandler = this.get('handleOutsideClick').bind(this);
 
-    ['click', 'touchend'].forEach((e) => document.addEventListener(e, clickHandler));
+    ['click', 'touchstart'].forEach((e) => document.addEventListener(e, clickHandler));
```
Comment (DaniFrancisco) [inline addon/mixins/click-outside.js]: did you check if there are any other components using this mixin? This might affect other stuff

Comment (migueldemoura) [inline addon/mixins/click-outside.js]: `touchend` is triggered when the user releases the finger after touching the screen, which is how we detect a "click". This mixin (`click-outside.js`) has that purpose.

**Review by migueldemoura — CHANGES_REQUESTED** [review]

**Review by AndreJoaquim — CHANGES_REQUESTED** [review]
I don't understand why do we need a scroll event listener on  *every* device to fix a bug on mobile... This should consider only mobile devices -- use the `media` service and add this only on mobile.

File: addon/components/uni-tooltip.js [hunk 3]
```diff
@@ -31,6 +31,18 @@ export default Component.extend(ClickOutsideMixin, {
     this._hideTooltip();
   },
 
+  didInsertElement() {
+    let clickHandler = this.get('handleOutsideClick').bind(this);
+
+    ['touchstart', 'scroll'].forEach((e) => document.addEventListener(e, clickHandler, { capture: true, passive: true }));
```
Comment (DaniFrancisco) [inline addon/components/uni-tooltip.js]: 120 chars

File: addon/components/uni-tooltip.js [hunk 4]
```diff
@@ -58,6 +70,7 @@ export default Component.extend(ClickOutsideMixin, {
   },
 
   _hideTooltip() {
+    console.log('KKKK');
```
Comment (DaniFrancisco) [inline addon/components/uni-tooltip.js]: remove

**Review by DaniFrancisco — CHANGES_REQUESTED** [review]

**Comment by niceoutput** [comment]
@AndreJoaquim This is for mobile and desktop, not mobile only. The title was updated.

**Review by DaniFrancisco — APPROVED** [review]

**Review by AndreJoaquim — APPROVED** [review]

**Review by migueldemoura — APPROVED** [review]

**Pull request merged**